pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -330,8 +259,40 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 					}
 				}
 			}
+			var prCursor *string
+			var pullRequests []PullRequestNode
+			var perPage int = 100
+
+			prCount, err := getPullRequestCount(ctx, genClient, name, ghs.cfg.GitHubOrg)
+			if err != nil {
+				ghs.logger.Sugar().Errorf("error getting pull request count", zap.Error(err))
+			}
+			ghs.logger.Sugar().Debugf("pull request count: %v for repo %v", prCount, repo)
 
-			ghs.getRepoPullRequestInformation(repoInfo)
+			prPages := getNumPages(float64(perPage), float64(prCount.Repository.PullRequests.TotalCount))
+			ghs.logger.Sugar().Debugf("pull request pages: %v for repo %v", prPages, repo)
+
+			for i := 0; i < prPages; i++ {
+				pr, err := getPullRequestData(ctx, genClient, name, ghs.cfg.GitHubOrg, perPage, prCursor)
+				if err != nil {
+					ghs.logger.Sugar().Errorf("error getting pull request data", zap.Error(err))
+				}
+
+				pullRequests = append(pullRequests, pr.Repository.PullRequests.Nodes...)
+
+				prCursor = &pr.Repository.PullRequests.PageInfo.EndCursor
+			}
+
+			for _, pr := range pullRequests {
+				createTime := pr.CreatedAt
+				closedTime := time.Now()
+				if pr.Closed {
+					closedTime = pr.ClosedAt
+				}
+
+				prAge := int64(closedTime.Sub(createTime).Hours())
+				ghs.mb.RecordGitRepositoryPullRequestTimeDataPoint(now, prAge, name, pr.HeadRefName)
```

Comment:
<img width="1562" alt="image" src="https://github.com/liatrio/liatrio-otel-collector/assets/25961386/6600084c-6b09-44cf-a2c6-fe99ee43a062">

It's a lot of data, i'm wondering if we should think through what needs to be a part of this metric as an opinion.

Reply:
Worst comes to worst I think we can average it across each repo since if we are trying to find patterns of working then it would accomplish mostly the same thing without needing to output all the fine grain data

Reply:
Agreed. I think my main concern here was that this visualization doesn't seem to have the attributes I was expecting it to. But I think i was overly tired last week so will recheck.

Reply:
Resolving this, the code looks fine with the attributes.